great_expectations/rule_based_profiler/types/data_assistant_result/volume_data_assistant_result.py

```diff
@@ -1,388 +1,11 @@
-            metric_type=metric_type,
-            plot_mode=plot_mode,
-            sequential=sequential,
-        )
-
-    def _create_return_chart_for_column_domain_expectation(
-        self,
-        expectation_configuration: ExpectationConfiguration,
-        attributed_metrics: Dict[Domain, Dict[str, ParameterNode]],
-        plot_mode: PlotMode,
-        sequential: bool,
-    ) -> alt.Chart:
-        domain: Domain
-        domains_by_column_name: Dict[str, Domain] = {
-            domain.domain_kwargs["column"]: domain
-            for domain in list(attributed_metrics.keys())
-        }
-
-        metric_configuration: dict = expectation_configuration.meta["profiler_details"][
-            "metric_configuration"
-        ]
-        domain_kwargs: dict = metric_configuration["domain_kwargs"]
-
-        domain = domains_by_column_name[domain_kwargs["column"]]
-
-        attributed_values_by_metric_name: Dict[str, ParameterNode] = attributed_metrics[
-            domain
-        ]
-
-        # Altair does not accept periods.
-        metric_name: str = list(attributed_values_by_metric_name.keys())[0].replace(
-            ".", "_"
-        )
-        metric_type: alt.StandardType = AltairDataTypes.QUANTITATIVE.value
-
-        df: pd.DataFrame = self._create_df_for_charting(
-            metric_name=metric_name,
-            attributed_values_by_metric_name=attributed_values_by_metric_name,
-            expectation_configuration=expectation_configuration,
-            plot_mode=plot_mode,
-        )
-
-        column_name: str = expectation_configuration.kwargs["column"]
-        subtitle = f"Column: {column_name}"
-
-        return self._chart_domain_values(
-            df=df,
-            metric_name=metric_name,
-            metric_type=metric_type,
-            plot_mode=plot_mode,
-            sequential=sequential,
-            subtitle=subtitle,
-        )
-
-    def _chart_column_values(
-        self,
-        column_dfs: List[Tuple[str, pd.DataFrame]],
-        metric_name: str,
-        metric_type: alt.StandardType,
-        plot_mode: PlotMode,
-        sequential: bool,
-    ) -> List[alt.VConcatChart]:
-        plot_impl: Callable[
-            [
-                List[Tuple[str, pd.DataFrame]],
-                str,
-                alt.StandardType,
-            ],
-            alt.VConcatChart,
-        ]
-        if plot_mode is PlotMode.PRESCRIPTIVE:
-            plot_impl = (
-                self.get_interactive_detail_expect_column_values_to_be_between_chart
-            )
-        else:
-            plot_impl = self.get_interactive_detail_multi_chart
-
-        display_chart: alt.VConcatChart = plot_impl(
-            column_dfs=column_dfs,
-            metric_name=metric_name,
-            metric_type=metric_type,
-            sequential=sequential,
-        )
-
-        return [display_chart]
+    # A mapping is defined for which metrics to plot and their associated expectations
+    EXPECTATION_METRIC_MAP = {
```

Comment:
@NathanFarmer let us add the type hint -- thanks.

Reply:
@alexsherstinsky adding a type hint will throw Mutable default ... is not allowed error due to this being a dataclass. I suspect this is why you did not type hint the ALLOWED_KEYS set directly below this constant.